docs/tutorials/jokes.md

```diff
@@ -1544,30 +1544,32 @@ So we've got a bit of extra work to do to avoid this development time problem.
 
 Note that this isn't a remix-only problem. Any time you have "live reload" of server code, you're going to have to either disconnect and reconnect to databases (which can be slow) or do the workaround I'm about to show you.
 
-💿 Copy this into a new file called `app/utils/db.server.ts`
+💿 Copy this into two new files called `app/utils/singleton.server.ts` & `app/utils/db.server.ts`
+
+```ts filename=app/utils/singleton.server.ts
+// Since the dev server re-requires the bundle, do some shenanigans to make
+// certain things persist across that 😆
+// Borrowed/modified from https://github.com/jenseng/abuse-the-platform/blob/2993a7e846c95ace693ce61626fa072174c8d9c7/app/utils/singleton.ts
+
+export const singleton = <Value>(
+  name: string,
+  valueFactory: () => Value
+): Value => {
+  const g = global as any;
+  g.__singletons ??= {};
+  g.__singletons[name] ??= valueFactory();
+  return g.__singletons[name];
+};
+```
```

Comment:
We now have [docs for how to do this](https://remix.run/docs/en/main/guides/manual-mode#keeping-in-memory-server-state-across-rebuilds), so could you update this code to be consistent with that and maybe link to those docs instead?

Reply:
This is the same as we have in `indie-stack`, where I just took over whatever @jenseng had in his code + adapted it to accept a function (just like Kent did in his personal website and the epic stack)

I would argue to bring the docs more in line with what we have in there tbh

Reply:
Ok, let's update the docs to use `singleton` as defined here, but then link to those docs here since the docs will have a more detailed explanation.